scad/utils.scad

```diff
@@ -1,5 +1,42 @@
 include <parameters.scad>
 
+/* Creates a flat 2D teardrop on the XY plane,
+ * where no angle in the positive y direction is greater than 45º. */
+module teardrop2d(radius){
+	// Find tangents on the circle at 45 degrees
+	// Radius is triangle hypotenuse
+	function tangent_point(circle_r, angle) = [
+		circle_r * cos(angle),
+				 circle_r * sin(angle),
+	];
+	teardrop_point = [0,
+	tangent_point(radius, 45).y + tangent_point(radius, 45).x];
+	circle(radius);
+	polygon([
+		tangent_point(radius, 45),
+		tangent_point(radius, 135),
+		teardrop_point
+	]);
+}
```

Comment:
Can you make the tangent angle configurable? I'm a little concerned about the tradeoff in wire retention

Reply:
Absolutely. I personally struggled to get the wires in, so I wrecked my retention off the bat with print cleanup and my mental model of how retention should work is bad. Can you say a little more about how you picture retention working and how it interacts with the teardrop?

Reply:
In my head the retention is a function of how much the channel walls overhang, which I think in your code is `cos(angle)`. (Personally I think it makes sense to define the angle as deflection from vertical, so 0 degrees is a vertical wall). Assuming wire depth stays constant, the angle is the main parameter that determines the size of the channel opening, and that just seems like something that folks might want to configure, depending on how much trouble they have getting the wires to fit into or stay in the channels.

Reply:
In my head the retention is a function of how much the channel walls overhang, which I think in your code is `cos(angle)`. (Personally I think it makes sense to define the angle as deflection from vertical, so 0 degrees is a vertical wall). Assuming wire depth stays constant, the angle is the main parameter that determines the size of the channel opening, and that just seems like something that folks might want to configure, depending on how much trouble they have getting the wires to fit into or stay in the channels.

Reply:
I think I kind of understand that. I'll make it customisable regardless, I think it makes sense even from the perspective that people tune printers to be able to do overhangs > 45º. 

Regarding retention (and I assume we're only talking about the uppermost channel here? (by default row)): This teardrop shape only gets wider than the equivalent cylinder once the tangent is at 45º, which I think is basically half way to a fully closed circle. I can't imagine anyone would have a wire falling out of something that narrow given the widest part of the circle is the wire diameter anyway. Does that sound right to you?